### PR TITLE
Add NaN test for sigmoid and grad of sigmoid

### LINKED ISSUE #94: Loss NaNs as a function of dataset size
Hello! Thanks for writing this great reimplementation of UniRep in JAX.

I'm running into a puzzling issue with the estimation of the holdout set loss in the fit function. I'm working on the Graham HPC cluster (https://docs.computecanada.ca/wiki/Graham) with 4 NVIDIA t4 chips, 16 GB RAM each. I'm trying to use the fit function on a dataset of about 50,000 sequences, all between 450 and 550 residues, using the random batching method.

I begin by shuffling the sequences, then taking the first 5,000 as holdout sequences and feeding the rest into the fit function, similarly to the workflow suggested in the docs.

When I fit a subset of <=11,000 sequences from the larger dataset of 50,000 sequences, I can set the learning rate to a reasonable 1e-5 and observe the loss dropping over several training epochs. However, as soon as I pass to the 'sequences' argument a character list of more than roughly 11,000 sequences, the evotuning log reports a loss value of NaN. I've found this threshold of ~11,000 sequences empirically.

I know there have been issues in the past with the final softmax layer producing NaNs and negative values, but I can't think of a reason the number of training sequences would factor into that. I would be grateful to hear any ideas or suggestions you might have.

Thanks!

### PR DESCRIPTION
**Pull request by ericmjl**
## PR Description

This PR closes #94. In there, @r-karimi uncovered a bug with NaN values showing up in the grad function. Upon deeper digging, I found out that NaN values could show up in the sigmoid function when large magnitude negative values are passed in. I thus modified the implementation of the sigmoid function such that we clip the value passed into sigmoid such that it is at minimum -88.0. 

### General

1. [x] I have made the PR off a new branch from my fork 
   (`<your_username>`:`<feature-branch_name>`), *not* 
   `<your_username>`:`master`.
2. [ ] I have added my changes to the `CHANGELOG.md` file at the top.
3. [x] I have made any necessary changes to the documentation in the `README`.

### Code checks
1. [x] If there are new features implemented, add suitable tests
   in the `tests` directory.
2. [-] If any new dependencies are introduced through the new features,
   add the packages, pinned to a version, to `environment.yml`.
3. [x] Run `make test` in a console in the top level directory
   to make sure all the tests pass.
4. [x] Run `make format` in a console in the top level directory
   to make the code comply with the formatting standards.

### CODE REVIEW

**Comment by ericmjl** [comment]
@ElArkk, after this PR, I think we should migrate the CI pipeline onto Azure DevOps. The economics are such that we shouldn't be using Travis CI anymore. (Only Azure DevOps gives us unlimited build minutes for open source projects.)

**Comment by ericmjl** [comment]
@ElArkk ready for review!

**Comment by ElArkk** [comment]
Looking good, merging!
